### PR TITLE
Fix/1751

### LINKED ISSUE #1751: Issue with nested delete operation relying on connectionAggregate count
**Describe the bug**

Let's say 
1. I have an application database which has multiple organizations. 
2. Each organization has its own admins. 
3. An admin can belong to multiple organizations as shown in the figure below.  

image

While deleting an organization, I would like to delete the organization together with its administrators as long as they don't participate in any other organization. 

In the mutation below, I try to delete the organization Google together with its admin(un named node). The other admins connected to Google (Mugukey and Mugumya) should not be deleted but only disconnected from Google. 

**Type definitions**

```graphql
type Organization {
    organizationId: ID! @id
    title: String
    createdAt: DateTime!
    admins: [Admin!]! @relationship(type: "HAS_ADMINISTRATOR", direction: OUT)
}

type Admin {
    adminId: ID! @id
    createdAt: DateTime!
    isSuperAdmin: Boolean
}
```

**To Reproduce**
Exexute this mutation

_Mutation_
```graphql
mutation DeleteOrganizations($where: OrganizationWhere, $delete: OrganizationDeleteInput) {
  deleteOrganizations(where: $where, delete: $delete) {
    nodesDeleted
    relationshipsDeleted
  }
}
```
_Variables_
```graphql
{
  "where": {
    "title": "Google"
  },
  "delete": {
    "admins": [
      {
        "where": {
          "node": {
            "organizationsAggregate": {
              "count": 1
            }
          }
        }
      }
    ]
  }
}
```
Upon running the operation above only the organization node Google gets deleted, and 3 relationships (1 from each of the admins)

**Expected behavior**
Deletion of the organization Google, deletion of the unnamed admin node attached to it, and deletion of the 3 relationships(1 from each of the attached admins).

**System**

* OS: [ Windows]
* Version: [@neo4j/graphql@^3.3.0]
* Node.js version: [v14.17.4]

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
Fix filtering on nested aggregation fields.
As a side effect, some parameters will be duplicated until a proper refactor is done to connections to remove nested parameters

Closes #1751

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at
 http://neo4j-graphql-toolbox-stage-1837.surge.sh (=> for local development; use for instance [Neo4j Desktop](https://neo4j.com/download/))
or
 https://neo4j-graphql-toolbox-stage-1837.surge.sh (=> use to connect to an [Aura Free](https://neo4j.com/cloud/aura-free/) dbms)

File: packages/graphql/src/translate/where/create-element-where-and-params.ts [hunk 1]
```diff
@@ -48,6 +50,7 @@ function createElementWhereAndParams({
     parameterPrefix: string;
     listPredicates?: ListPredicate[];
 }): [string, any] {
+    // ADD agregation where here
```
Comment (darrellwarde) [inline packages/graphql/src/translate/where/create-element-where-and-params.ts:53]: I'm assuming this is leftover? 👀

**Review by darrellwarde — APPROVED** [review]
LGTM!

**Comment by darrellwarde** [comment]
Disregard, missing a changeset!!!

**Comment by angrykoala** [comment]
The pipeline error seems to be a nested dependency (`yargs-parser`). I'll merge as it is not caused by this PR

**Pull request merged**